### ISSUE TITLE
MLP forces softmax when output is more than 1D

### ISSUE THREAD

**Issue by martelloni** [MusicallyEmbodiedML/MLP_XMOS#3]
`MLP` line 179: softmax should not be applied to all outputs. Also, softmax is not backpropagated. Remove and handle the case externally (if needed or supported).
